docs/integrations/airflow/preflight-check-dag.md

```diff
@@ -0,0 +1,332 @@
+        except ImportError as e:
+            raise ValueError("OpenLineage provider is not accessible") from e
+    else:
+        try:
+            from openlineage.airflow.plugin import OpenLineagePlugin as plugin
+        except ImportError as e:
+            raise ValueError("OpenLineage is not accessible") from e
+
+    if len(plugin.listeners) == 1:
+        return True
+
+    return False
+
+
+def _is_ol_disabled():
+    if _provider_can_be_used():
+        try:
+            # apache-airflow-providers-openlineage >= 1.7.0
+            from airflow.providers.openlineage.conf import is_disabled
+        except ImportError:
+            # apache-airflow-providers-openlineage < 1.7.0
+            from airflow.providers.openlineage.plugins.openlineage import _is_disabled as is_disabled
+    else:
+        from openlineage.airflow.plugin import _is_disabled as is_disabled
+
+    if is_disabled():
+        if _provider_can_be_used() and conf.getboolean("openlineage", "disabled", fallback=False):
+            raise ValueError("OpenLineage is disabled in airflow.cfg: openlineage.disabled")
+        elif os.getenv("OPENLINEAGE_DISABLED", "false").lower() == "true":
+            raise ValueError(
+                "OpenLineage is disabled due to the environment variable OPENLINEAGE_DISABLED"
+            )
+        raise ValueError(
+            "OpenLineage is disabled because required config/env variables are not set. "
+            "Please refer to "
+            "https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/guides/user.html"
+        )
+    return False
+
+
+def _get_transport():
+    if _provider_can_be_used():
+        from airflow.providers.openlineage.plugins.openlineage import OpenLineageProviderPlugin
+        transport = OpenLineageProviderPlugin().listeners[0].adapter.get_or_create_openlineage_client().transport
+    else:
+        from openlineage.airflow.plugin import OpenLineagePlugin
+        transport = (
+            OpenLineagePlugin.listeners[0].adapter.get_or_create_openlineage_client().transport
+        )
+    return transport
+
+def is_ol_accessible_and_enabled():
+    if not _is_listener_accessible():
+        _is_ol_disabled()
+
+    try:
+        transport = _get_transport()
+    except Exception as e:
+        raise ValueError("There was an error when trying to build transport.") from e
+
+    if transport is None or transport.kind in ("noop", "console"):
+        _debug_missing_transport()
+
+
+def validate_connection():
+    transport = _get_transport()
+    config = attr.asdict(transport.config)
+    verify_backend(LINEAGE_BACKEND, config)
+
+
+def verify_backend(backend_type: str, config: dict):
+    backend_type = backend_type.lower()
+    if backend_type == "marquez":
+        return _verify_marquez_http_backend(config)
+    elif backend_type == "atlan":
+        return _verify_atlan_http_backend(config)
+    elif backend_type == "marquez":
```

Comment:
Probably just `else:`

Reply:
Thanks! this got missed 😅

Reply:
Not sure i understand, where do we need `else` here? We should probably change "marquez" to "custom" tho, as it's duplicated now ;)